Mathlib/Algebra/Group/Prod.lean

```diff
@@ -737,69 +716,39 @@ variable {P : Type*} [Monoid P]
   with a commutation assumption:
   `f.noncommCoprod g _ (p : M × N) = f p.1 * g p.2`.
   (Noncommutative case; in the commutative case, use `MonoidHom.coprod`.)-/
-@[to_additive
+@[to_additive (attr := simps)
     "Coproduct of two `AddMonoidHom`s with the same codomain,
     with a commutation assumption:
     `f.noncommCoprod g (p : M × N) = f p.1 + g p.2`.
     (Noncommutative case; in the commutative case, use `AddHom.coprod`.)"]
 def noncommCoprod : M × N →* P where
   toFun := fun mn ↦ (f mn.fst) * (g mn.snd)
-  map_mul' := fun mn mn' ↦ by
-    simp only [Prod.fst_mul, Prod.snd_mul, map_mul]
-    simp only [mul_assoc]
-    apply congr_arg₂ _ rfl
-    simp only [← mul_assoc]
-    apply congr_arg₂ _ _ rfl
-    apply comm
-  map_one' := by
-    simp only [Prod.fst_one, Prod.snd_one, map_one, mul_one]
-
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply (p : M × N) :
-  f.noncommCoprod g comm p = f p.1 * g p.2 :=
-  rfl
-
-/- -- simp_normal_form complains
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply' (p : M × N) :
-    f.noncommCoprod g comm p = g p.2 * f p.1 :=
-  comm p.1 p.2
--/
+  map_one' := by simp only [Prod.fst_one, Prod.snd_one, map_one, mul_one]
+  __ := f.toMulHom.noncommCoprod g.toMulHom comm
 
 @[to_additive (attr := simp)]
 theorem noncommCoprod_comp_inl : (f.noncommCoprod g comm).comp (inl M N) = f :=
-  ext fun x => by simp [noncommCoprod_apply]
+  ext fun x => by simp
 
 @[to_additive (attr := simp)]
 theorem noncommCoprod_comp_inr : (f.noncommCoprod g comm).comp (inr M N) = g :=
-  ext fun x => by simp [noncommCoprod_apply]
-
-@[to_additive]
-theorem noncommCoprod.commute (f : M × N →* P) (m : M) (n : N) :
-    Commute (f.comp (inl M N) m) (f.comp (inr M N) n) := by
-  simp only [coe_comp, Function.comp_apply]
-  apply Commute.map
-  apply inl_inr_commute
```

Comment:
I removed this lemma, as the name was a bit strange, the lemma isn't in simpNF (if it were, the `comp` would be eliminated) and `(inl_inr_commute _ _).map f` is a pretty short proof.